### PR TITLE
Change logics for code generation for code attributes #2570

### PR DESCRIPTION
**Pull request by KartnDev**
image

### CODE REVIEW

File: jmix-dynattr/dynattr/src/main/java/io/jmix/dynattr/utils/DynAttrStringUtils.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    private DynAttrStringUtils() {
+    }
+
+    /**
+     * Converts a string to camelCase format.
+     * <p>
+     * This method ensures that the camelCase or PascalCase format is preserved when transforming a string into camelCase.
+     * For example, 'camelCase' will not be transformed into 'camelcase'.
+     *
+     * @apiNote This method is designed to prevent unintentional conversion of camelCase or PascalCase strings into lowercase.
```
Comment (gorbunkov) [inline jmix-dynattr/dynattr/src/main/java/io/jmix/dynattr/utils/DynAttrStringUtils.java:46]: What's the purpose of the `@apiNote` tag? I never saw it and I don't get why not to write this text as a new paragraph?

File: jmix-dynattr/dynattr/src/main/java/io/jmix/dynattr/utils/DynAttrStringUtils.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        if (Strings.isNullOrEmpty(str) || isCamelCase(str)) {
+            return str;
+        } else if (isPascalCase(str)) {
+            return StringUtils.uncapitalize(str);
+        } else {
+            return CaseUtils.toCamelCase(str, false, delimiters);
+        }
+    }
+
+
```
Comment (gorbunkov) [inline jmix-dynattr/dynattr/src/main/java/io/jmix/dynattr/utils/DynAttrStringUtils.java:63]: Too many empty lines between methods

File: jmix-dynattr/dynattr/src/test/groovy/dynamic_attributes/DynAttrStringUtilsTest.groovy [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dynamic_attributes
+
+import io.jmix.dynattr.utils.DynAttrStringUtils
+import spock.lang.Specification
+
+class DynAttrStringUtilsTest extends Specification {
```
Comment (gorbunkov) [inline jmix-dynattr/dynattr/src/test/groovy/dynamic_attributes/DynAttrStringUtilsTest.groovy:22]: I don't see tests for input that consists of several words, e.g. "Some text" -> someText or "some text" -> someText

**Review by gorbunkov — CHANGES_REQUESTED** [review]

**Comment by KartnDev** [comment]
Add "space cases" tests
image

File: jmix-dynattr/dynattr/src/test/groovy/dynamic_attributes/DynAttrStringUtilsTest.groovy [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+        when:
+        String convertedCamelCase = DynAttrStringUtils.toCamelCase(pascalCaseString, delimiters)
+
+        then:
+        convertedCamelCase == "spaceCaseString"
+    }
+
+    def "toCamelCase in half-capitalized Space case"() {
+        given:
+        String pascalCaseString = "Space case String"
```
Comment (gorbunkov) [inline jmix-dynattr/dynattr/src/test/groovy/dynamic_attributes/DynAttrStringUtilsTest.groovy:93]: All such variables in all tests are called `pascalCaseString` that doesn't seem correct. Rename these variables.

**Review by gorbunkov — CHANGES_REQUESTED** [review]

**Review by gorbunkov — APPROVED** [review]

**Pull request merged**